spec/integration/application/agent_spec.rb

```diff
@@ -326,4 +326,75 @@
       end
     end
   end
+
+  context 'multiple agents running' do
+    it "exits if an agent is already running" do
+      server.start_server do |port|
+        path = Puppet[:agent_catalog_run_lockfile]
+        Puppet[:masterport] = port
+
+        th = Thread.new {
+          %x{ruby -e "File.write('#{path}', Process.pid); sleep(2); 'puppet keyword needed here'"}
```

Comment:
The `puppet keyword ...` is because our stale lock detection code requires ['puppet' to be in the process args](https://github.com/puppetlabs/puppet/blob/ee4ec4a511965be44ef17f4251aad1270fdf164e/lib/puppet/util/pidlock.rb#L90) on posix. It might be more obvious to change the process title instead:

```ruby
%x{ruby -e "$0 = 'puppet'; File.write('#{path}', Process.pid); sleep(2)"}
```

Reply:
addressed in https://github.com/puppetlabs/puppet/pull/8103